01-data-structures/05-hashes-part-2/open_addressing/open_addressing.rb

```diff
@@ -2,29 +2,123 @@
 
 class OpenAddressing
   def initialize(size)
+    @items = Array.new(size)
+    @item_count = 0
   end
 
   def []=(key, value)
+    puts "-" * 10
+    puts "Entry is #{key}"
+    hash_index = index(key, @items.size)
+    node = Node.new(key, value)
+
+    if @items[hash_index] == nil
+      puts "Inserting #{key} at #{hash_index}"
+      @items[hash_index] = node
+      @item_count += 1
+    # elsif @items[hash_index].key == key && next_open_index(hash_index) != -1
+    #   return
+    # elsif @items[hash_index].key == key
+    #   self.resize
+    #   return
+    else
+      open_index = next_open_index(hash_index)
+      if open_index == -1
+        puts "Resizing the array"
+        self.resize
+        self[key] = value
+        return
+      else
+        puts "Found open index at #{open_index}"
+        @items[open_index] = node
+        @item_count += 1
+      end
+    end
   end
 
   def [](key)
+    hash_index = index(key, @items.size)
+    search_index = 0
+    puts "Checking #{hash_index} for #{key}"
+
+    if @items[hash_index] == nil
+      return nil
+    elsif @items[hash_index].key == key
+      puts "Found at #{hash_index}"
+      return @items[hash_index].value
+    else
+      hash_index == @items.size - 1 ? search_index = 0 : search_index = hash_index + 1
```

Comment:
I prefer to have the assignment outside of ternaries:

    search_index = hash_index == @items.size - 1 ? 0 : hash_index + 1

In doing so, you can see a potential shortcut that (I personally feel) is easier to read than a ternary:

    search_index = (hash_index + 1) % @items.size